aten/src/ATen/native/native_functions.yaml

```diff
@@ -13163,11 +13163,23 @@
     CPU, CUDA, NestedTensorCPU, NestedTensorCUDA: native_multi_head_attention
   autogen: _native_multi_head_attention.out
 
-- func: _scaled_dot_product_attention(Tensor query, Tensor key, Tensor value, Tensor? attn_mask=None, float dropout_p=0.0, bool need_attn_weights=True, bool is_causal=False) -> (Tensor, Tensor)
+- func: _scaled_dot_product_attention(Tensor query, Tensor key, Tensor value, Tensor? attn_mask=None, float dropout_p=0.0, bool need_attn_weights=False, bool is_causal=False) -> (Tensor, Tensor)
   python_module: nn
   variants: function
   autogen: _scaled_dot_product_attention.out
 
+# Register the math kernel for cpu
+- func: _scaled_dot_product_attention_forward(Tensor query, Tensor key, Tensor value, Tensor? attn_mask=None, float dropout_p=0.0, bool need_attn_weights=False, bool is_causal=False) -> (Tensor, Tensor)
+  variants: function
+  dispatch:
+    CUDA: _scaled_dot_product_attention_forward_cuda
+    CPU: _scaled_dot_product_attention_forward_math
```

Comment:
`_scaled_dot_product_attention_forward_math` is the same as _scaled_dot_product_attention_math. Since _scaled_dot_product_attention jumps straight to forward if it's inputs don't requires grad. We could add more checks at this layer but this would bubble up some logic out of the forward function which I think is undesirable.